presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -147,6 +150,8 @@
     // for recursive view detection
     private final Deque<Table> tablesForView = new ArrayDeque<>();
 
+    private final Map<Table, Integer> tablesForMV = new HashMap<>();
```

Comment:
Can we use something similar to `tablesForView` with a `Deque`? A Deque is good enough as we are using depth-first search.

Actually, this might not be needed at all. view could be nested but materialized view is guaranteed to be not nested. So if we visit table again inside processing the materialized view, it is not going to come back again.

Reply:
The nesting would not be the user generated. The nesting is generated as per our design.
Once we observe a MV, we replace it with new query having an MV again. So when analyzer would see the MV again in a new query. Without this protection, the MV expansion would be recursive.

Reply:
Let's use `private final Deque<Table> tablesForMaterializedView = new ArrayDeque<>();` to be consistent. We don't have to do the counting. Just check if it contains the view or not to prevent recursion.

Reply:
This wont work, we do want one view to be processed 2 times. I will come up with better terminology, rather than table counting. something like IsMaterializedViewExpanded